From c1906694674b9223f1ff1a3ac80f905aefb0b800 Mon Sep 17 00:00:00 2001
From: Thomas Bonnefille <thomas.bonnefille@bootlin.com>
Date: Fri, 17 Jan 2025 22:10:41 +0100
Subject: [PATCH] Replace cuserid

cuserid has been removed from some toolchains and isn't in the POSIX
standard anymore since 1990. getlogin has been in the standard since
2001.

This commit uncomment the call to getlogin and so replace cuserid with
getlogin.

Signed-off-by: Thomas Bonnefille <thomas.bonnefille@bootlin.com>
---
 pklock.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/pklock.c b/pklock.c
index dc30b76..5f4865b 100644
--- a/pklock.c
+++ b/pklock.c
@@ -80,8 +80,7 @@ char *dolock(char *fname)
 	}
 	if ((n = read(fd, locker, MAXNAME)) < 1) {
 		lseek(fd, 0, SEEK_SET);
-/*		strcpy(locker, getlogin()); */
-		cuserid(locker);
+		strcpy(locker, getlogin());
 		strcat(locker + strlen(locker), "@");
 		gethostname(locker + strlen(locker), 64);
 		write(fd, locker, strlen(locker));
-- 
2.48.1

